staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_dispatcher.go

```diff
@@ -76,7 +72,7 @@ type PolicyInvocation[P runtime.Object, B runtime.Object, E Evaluator] struct {
 //
 // The delegate provides the "validation" or "mutation" aspect of dispatcher functionality
 // (in contrast to generic.PolicyDispatcher which only selects active policies and params)
-type dispatcherDelegate[P, B runtime.Object, E Evaluator] func(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces, versionedAttributes webhookgeneric.VersionedAttributeAccessor, invocations []PolicyInvocation[P, B, E]) error
+type dispatcherDelegate[P, B runtime.Object, E Evaluator] func(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces, versionedAttributes webhookgeneric.VersionedAttributeAccessor, invocations []PolicyInvocation[P, B, E]) ([]PolicyError, *apierrors.StatusError)
```

Comment:
While trying to understand the dispatcher concept I noticed that it is only used in mutating admission, but was preexisting before the PR? `NewPolicyDispatcher` not called from the validating code. Why is that?

Reply:
Yes, it looks like validating admission retains it's own unshared dispatcher.  This might to insulate ValidatingAdmissionPolicy from the changes in this PR?

cc @alexzielenski for any additional context/history

Reply:
The VAP dispatcher embeds the generic dispatcher. There are separate dispatchers because for validating vs mutating case the actual evaluation of what the hook does is different. The validating dispatcher evaluates the policies against the request and returns an error without modifying anything. The mutating dispatcher will change the object according to the policy. But both have shared code in `generic.Dispatcher` used for corralling all the policies, bindings, params, etc.

Reply:
Thanks @alexzielenski. We can follow up on the code sharing in follow up PRs.